src/platform/packages/private/kbn-grid-layout/grid/grid_row/grid_row_header.tsx

```diff
@@ -67,9 +82,46 @@ export const GridRowHeader = React.memo(
           setPanelCount(count);
         });
 
+      const dragRowStyleSubscription = gridLayoutStateManager.activeRow$
+        .pipe(
+          pairwise(),
+          map(([before, after]) => {
+            if (!before && after) {
+              return { type: 'init', activeRow: after };
+            } else if (before && after) {
+              return { type: 'update', activeRow: after };
+            } else {
+              return { type: 'finish', activeRow: before };
+            }
```

Comment:
Not sure if this is actually more efficient - in my testing, the impact was minimal. But in theory, this would be much more efficient when we have a **lot** of sections, because it prevents us from setting styles unnecessarily on sections that are not being updated.